sjb/config/test_cases/ami_build_origin_int_rhel_fork.yml

```diff
@@ -0,0 +1,276 @@
+        fi;
+        popd >/dev/null"
+        ssh -F ./.config/origin-ci-tool/inventory/.ssh_config openshiftdevel sudo GOPATH=/data go get golang.org/x/tools/cmd/cover
+      }
+
+      if [[ "${TARGET_ORIGIN_BRANCH}" =~ enterprise-3.[1-2] || "${TARGET_ORIGIN_BRANCH}" =~ release-1.[1-2] ]]; then
+        install_go_1_4
+        export OS_BUILD_ENV_IMAGE="openshift/origin-release:golang-1.4"
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" =~ enterprise-3.[3-4] || "${TARGET_ORIGIN_BRANCH}" =~ release-1.[3-4] ]]; then
+        oct prepare golang --version=1.6
+        export OS_BUILD_ENV_IMAGE="openshift/origin-release:golang-1.6"
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" =~ enterprise-3.5 || "${TARGET_ORIGIN_BRANCH}" =~ release-1.5 ]]; then
+        oct prepare golang --version=1.7.5 -u https://cbs.centos.org/repos/paas7-openshift-origin36-candidate/x86_64/os/
+        export OS_BUILD_ENV_IMAGE="openshift/origin-release:golang-1.7"
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" == "master" ]]; then
+        oct prepare golang --version=1.7.5 -u https://cbs.centos.org/repos/paas7-openshift-origin36-candidate/x86_64/os/
+        export OS_BUILD_ENV_IMAGE="openshift/origin-release:golang-1.7"
+
+      else
+        echo "[FATAL] No Golang version is known for \`${TARGET_ORIGIN_BRANCH}\`.
+              Contact Steve Kuznetsov <skuznets@redhat.com> for assistance."
+        exit 1
+      fi
+  - type: "host_script"
+    title: "install docker"
+    script: |-
+      if [[ "${TARGET_ORIGIN_BRANCH}" == "enterprise-3.1" || "${TARGET_ORIGIN_BRANCH}" == "release-1.1" ]]; then
+        oct prepare docker --version=1.8.2
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" == "enterprise-3.2" || "${TARGET_ORIGIN_BRANCH}" == "release-1.2" ]]; then
+        oct prepare docker --version=1.9.1 --repo=oso-rhui-rhel-server-*
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" == "enterprise-3.2.1" || "${TARGET_ORIGIN_BRANCH}" == "release-1.2.1" ||
+              "${TARGET_ORIGIN_BRANCH}" == "enterprise-3.3" || "${TARGET_ORIGIN_BRANCH}" == "release-1.3" ]]; then
+        oct prepare docker --version=1.10.3 --repo=oso-rhui-rhel-server-*
+          
+      elif [[ "${TARGET_ORIGIN_BRANCH}" =~ enterprise-3.[4-5] || "${TARGET_ORIGIN_BRANCH}" =~ release-1.[4-5] ]]; then
+        oct prepare docker --version=1.12.6 --repo=oso-rhui-rhel-server-* -u https://mirror.openshift.com/enterprise/rhel/dockerextra/x86_64/os/
+
+      elif [[ "${TARGET_ORIGIN_BRANCH}" == "master" ]]; then
+        oct prepare docker --version=1.12.6 --repo=oso-rhui-rhel-server-* -u https://mirror.openshift.com/enterprise/rhel/dockerextra/x86_64/os/
+
+      else
+        echo "[FATAL] No Docker version is known for \`${TARGET_ORIGIN_BRANCH}\`.
+              Contact Steve Kuznetsov <skuznets@redhat.com> for assistance."
+        exit 1
+      fi
+  - type: "host_script"
+    title: "prepare repositories"
+    script: |-
+      oct prepare repositories
+      # Replace origin repo with the specified fork
```

Comment:
Although if this is a private repo (fork of `ose`) there will not be a key to do it on the remote machine -- so we'd need to clone the correct state locally then use 

```sh
oct sync local origin --branch "${ORIGIN_BRANCH}" --merge-into master
```